src/index.ts

```diff
@@ -120,41 +120,28 @@ const index = ({
 
   // bed offset Z logic:
   if (palette) {
+    /* For multi-material project, we use the highest Z offset
+      among all materials used in the first layer.
+      This logic is handled in a post-processing script.
+      Generate the `zOffsetPerExt` array to store the Z offset for each material.
+    */
     for (let i = 0; i < extruderCount; i++) {
       if (drivesUsed[i]) {
         const zOffset = getMaterialFieldValue(materials[i]!, 'zOffset', style.zOffset);
         profile.zOffsetPerExt[i] = zOffset;
       }
     }
+    // Set `profile.zOffset` to 0 to avoid unnecessary calculations in the post-processing script.
     profile.zOffset = 0;
   } else {
-    // - ignore inputs we know we won't be using
-    // - only look at project's z-offset setting if no inputs have a material override
-    // - use the highest z-offset seen (including 0)
-    let useZOffsetFromStyle = true;
-    for (let i = 0; i < extruderCount; i++) {
-      if (drivesUsed[i]) {
-        const material = materials[i]!;
-        if (material.style.useZOffset) {
-          useZOffsetFromStyle = false;
-          break;
-        }
-      }
-    }
-    let zOffset = -Infinity;
-    if (useZOffsetFromStyle) {
-      zOffset = style.zOffset ?? 0;
+    // Assume the project has only one input for a "non-palette" project
+    // (i.e., it is not an element and does not use a palette).
+    const material = materials[0]!;
```

Comment:
The assumption that only one input is used is correct, but `palette` being falsy doesn't mean that the project only has one input:
- e.g. a project with a Palette 3 Pro (8 inputs)
- `drivesUsedHint` only lists one drive used, so we downgrade the workflow to standard
- the project data still has 8 inputs though

As I mentioned yesterday, you can find the actual input that's used easily:
```ts
// if no Palette, then only one input is used in the project
// but it may not be input 0
const singleExtUsed = drivesUsed.findIndex((idx) => idx);
const material = materials[singleExtUsed]!;
```